### PR TITLE
Proposal to discuss

### PR DESCRIPTION
**Pull request by zverok**
Hey. 
I know I am pretty late to the party, but I hope at least some of my suggestions could be helpful.

First of all: thanks for all the impressive work!
Second: this PR is _not_ for merge, neither it should be treated as an instruction "do this/do that".
It is just so that GitHub PR interface is pretty convenient to show and discuss inplace some things that I feel as important.

So, just read it and do whatever you feel right :) (I'll write some comments to PR lines, please wait for them.)

### CODE REVIEW

File: app/controllers/sciruby_github_controller.rb [hunk 1]
```diff
@@ -8,27 +8,11 @@ def repos
...
-    }
-    @df_repo_table = Daru::View::Table.new(@df_repos, table_opts)
-    @df_repos_col_chart = Daru::View::Plot.new(@df_repo_table, opts)
-    opts = {
-      type: :histogram,
-      adapter: :googlecharts,
-      height: 500,
-      width: 1000
-    }
-    @df_repos_histo_chart = Daru::View::Plot.new(@df_repo_table, opts)
```
Comment (zverok) [inline app/controllers/sciruby_github_controller.rb:29]: To understand, why I feel all of it is not necessary here, please refer to view (repos.erb)

File: app/controllers/sciruby_github_controller.rb [hunk 2]
```diff
@@ -8,27 +8,11 @@ def repos
...
+    respond_to do |format|
+      format.html
 
-    export(@df_repos)
+      Daru::View::Helpers.respond(format, @df_repos, formats: %w[csv xls json])
```
Comment (zverok) [inline app/controllers/sciruby_github_controller.rb:14]: I am not sure about it (especially syntax), but I feel like task "this controller responds to those types, and returns exported dataframe" should be one-liner somehow, solved by daru-view, not by library user.

File: app/views/sciruby_github/repos.erb [hunk 3]
```diff
@@ -2,18 +2,38 @@
 <div class="span12 pagination-centered" style="text-align: center;">
     <h3> SciRuby github data analysis (Using daru-view/googlecharts)</h3>
     <b>Data is taken from url : https://api.github.com/orgs/Sciruby/repos </b>
-    <%= render 'export' %>
+    <%=
+      daru_download formats: %w[csv xls json], separator: '|'
```
Comment (zverok) [inline app/views/sciruby_github/repos.erb:6]: The same idea as in controller: "just give me damn button!"

File: app/views/sciruby_github/repos.erb [hunk 4]
```diff
@@ -2,18 +2,38 @@
...
+    %>
     <p style="left: 50%">
       <b>SciRuby Datatable : </b>
-      <%=raw @df_repo_table.div %>
+      <%=
+        daru_table @df_repos,
+                   adapter: :googlecharts,
+                   pageSize: 10,
+                   height: 300,
+                   width: 400
```
Comment (zverok) [inline app/views/sciruby_github/repos.erb:15]: This (and `daru_chart` below) seems for me:
a) pretty important
b) pretty easy to implement (just what you already have + small helper method).

Ideas here:
1. when Rails guy sees something like `raw something`, it feels like a dirty hack.
2. when tutorial says that to show something in view, you should prepare it in controller, it is "wrong" concern separation
3. when "view-related" options are stored in controller (width, height, and so on) -- it also feels wrong (imagine there are some header image above, which should be exactly the width of the table; now, you replace image with another one and think "Oh, I need to make table 50px wider... I'll just fix my view. What??? Where is this width param?.."). 
4. this way (helper method in view) feels most "natural": "OK, I've added some _view_ library, now I have some helper in _view_ and here is its params".

**Review by zverok — COMMENTED** [review]
Here are the comments! Feel free to use them any way you want ;)

File: app/controllers/sciruby_github_controller.rb (hunk 2 above)
Comment (athityakumar) [reply app/controllers/sciruby_github_controller.rb:14]: Acknowledged. This is already under discussion at https://github.com/Shekharrajak/daru-view/issues/53

File: app/views/sciruby_github/repos.erb (hunk 4 above)
Comment (Shekharrajak) [reply app/views/sciruby_github/repos.erb:15]: Thanks for the ideas. 

We can have `daru_chart` , `daru_table` helpers for rails (refer [this](https://github.com/winston/google_visualr/blob/master/lib/google_visualr/app/helpers/view_helper.rb#L13) how `render_chart` can be used in rails app. Similarly we can have  `daru_table`, `daru_chart` ) , Sinatra (refer [sinatra helpers](https://www.sitepoint.com/using-sinatra-helpers-to-clean-up-your-code/)  and [sinatra_extension](http://www.sinatrarb.com/extensions.html) that seems helpful for it ),  not sure about how similar things can be done in Nanoc web applications.

File: app/views/sciruby_github/repos.erb (hunk 3 above)
Comment (Shekharrajak) [reply app/views/sciruby_github/repos.erb:6]: Thinking about it. Thanks for idea!

File: app/views/sciruby_github/repos.erb (hunk 4 above)
Comment (zverok) [reply app/views/sciruby_github/repos.erb:15]: I believe (considering how small time we have left) that just doing the "most popular" thing (e.g., for now, just add helper for Rails, to showcase everything properly) would be enough for this week ;)

**Comment by Shekharrajak** [comment]
@zverok, please  have a look in this PR https://github.com/Shekharrajak/daru_examples_io_view_rails/pull/13 where I am using `daru_chart` and `daru_table`.

live demo : 

https://daru-examples-io-view-rails.herokuapp.com/sciruby/watchers
https://daru-examples-io-view-rails.herokuapp.com/sciruby/repos

**Comment by Shekharrajak** [comment]
#13 , is using the changes of https://github.com/Shekharrajak/daru-view/pull/55

**Pull request closed without merging**